stimuli.js

```diff
@@ -142,8 +142,116 @@ function pickRandomList() {
         return empty_array;
     }
     let num_lists = TEST_ITEMS.length;
-    var shuffled_range = jsPsych.randomization.repeat(range(num_lists), 1)
-    var retlist = TEST_ITEMS[shuffled_range[0]];
+    let shuffled_range = jsPsych.randomization.repeat(range(num_lists), 1)
+    let retlist = TEST_ITEMS[shuffled_range[0]];
     return retlist
 }
 
+/**
+ * Checks whether a stimulus contains precisely 1 target stimulus
+ *
+ * @param {object} trial
+ *
+ * @return {boolean}  true if the trial contains precisely one target.
+ */
+function containsOneTarget(trial) {
+    let sum = 0;
+    let has_visual =   typeof trial.visual_target === "string" &&
+                       trial.visual_target.length > 0;
+    let has_auditory = typeof trial.auditory_target === "string" &&
+                       trial.auditory_target.length > 0;
+    if (has_visual)
+        sum += 1;
+    if (has_auditory)
+        sum += 1;
+    return sum === 1;
+}
+
+/**
+ * Checks whether the trial doesn't contain more than 1 prime.
+ * @param trial
+ * @return {boolean}
+ */
+function containsAtMostOnePrime(trial) {
+    let sum = 0;
+    let visprime = trial.visual_prime && trial.visual_prime.length > 0;
+    let audprime = trial.auditory_prime && trial.auditory_prime.length > 0;
+    if (visprime) {
+        sum += 1;
+    }
+    if (audprime) {
+        sum += 1;
+    }
+    return sum <= 1;
+}
+
+/**
+ * Validates a list of stimuli.
+ *
+ * The list_name parameters is mostly used to ease the lookup of an invalid
+ * items. The name is used together with the item id to print the violating
+ * items.
+ *
+ * @param {[{}]} trials A list with trial parameters
+ * @param {string} list_name The name of the list.
+ *
+ * @return {boolean} returns true when no errors are found, false otherwise
+ */
+function validateStimuli(trials, list_name) {
+    let success = true;
+    trials.forEach((trial) => {
+
+        let stim_string =
+            `The stimulus with id "${trial.id}" in list "${list_name} "`;
+
+        if (!containsOneTarget(trial)) {
+            console.error(stim_string + "does not contain precisely 1 target.");
+            success = false;
+        }
+        if (!containsAtMostOnePrime(trial)) {
+            console.error(stim_string + "does not contain at most 1 prime.");
+            success = false;
+        }
+    });
+    return success;
+}
+
+function validateAllStimuli() {
+    let success = true;
+    if (!validateStimuli(PRACTICE_LIST, "practice items")) {
+        success = false;
+    }
+    for (let i = 0; i < TEST_ITEMS.length; i++) {
+        let item = TEST_ITEMS[i];
+        if (!validateStimuli(item.table, item.list_name))
+            success = false;
+    }
+    return success;
+}
+
+/**
+ * Extracts all auditory stimuli from the trials.
+ *
+ * This function makes it somewhat easy to preload the auditory stimuli
+ *
+ * @return {string[]}
+ */
+function gatherAudioStimuli() {
```

Comment:
I don't see this being used, is that coming in another PR?

Reply:
Yes, when working on the stimulus validation I took this along. It's related to #5